uwblueprint/focus-on-nature
# Camp Overview: Edit Camper Information

## Notion ticket link

[Camp Overview: Edit Camper Information](https://www.notion.so/uwblueprintexecs/Task-Board-F22-9e03bf9f7c4343f09a8a5c16dc48a12a?p=e1dd8cd472ef4ad6a2de3cf1026e01fa&pm=c)



## Implementation description
* Updated the updateCampersByID backend route + function to only update the fields of a camper that can be updated by the client
* Made a folder called `EditCamperModal` which has all the files needed for the edit camper modal functionality. 
* Made a type for editCamperModal which is used by the different files to pass around props 



## Steps to test
1. Go to a camp overview page (http://localhost:3000/admin/camp/63139c7bc3d7b55b44a01531) and click on the hamburger icon next to a registered camper (use mine if you want)
2. Click on the edit button and a modal with all the current info should appear 
3. Try to change all the fields to make sure the results get updated as expected
4. The page should reload after 3 seconds (this might not be the exact functionality we want but the alternative was taking too long to implement for now)



## What should reviewers focus on?
* I removed a lot of code from the backend so just make sure that looks ok 
* Currently, I am refreshing the page after 3 seconds so that the new camper info is loaded after edit. Not sure if this is the best functionality. Ideally we use a global state but I didn't have time to implement that


## Checklist
